src/vtquery.cpp

```diff
@@ -360,60 +360,71 @@ struct Worker : Nan::AsyncWorker {
     }
 
     void HandleOKCallback() override {
-        Nan::HandleScope scope;
-
-        v8::Local<v8::Object> results_object = Nan::New<v8::Object>();
-        v8::Local<v8::Array> features_array = Nan::New<v8::Array>();
-        results_object->Set(Nan::New("type").ToLocalChecked(), Nan::New<v8::String>("FeatureCollection").ToLocalChecked());
-
-        // for each result object
-        while (!results_queue_.empty()) {
-            auto const& feature = results_queue_.back(); // get reference to top item in results queue
-            if (feature.distance < std::numeric_limits<double>::max()) {
-                // if this is a default value, don't use it
-                v8::Local<v8::Object> feature_obj = Nan::New<v8::Object>();
-                feature_obj->Set(Nan::New("type").ToLocalChecked(), Nan::New<v8::String>("Feature").ToLocalChecked());
-
-                // create geometry object
-                v8::Local<v8::Object> geometry_obj = Nan::New<v8::Object>();
-                geometry_obj->Set(Nan::New("type").ToLocalChecked(), Nan::New<v8::String>("Point").ToLocalChecked());
-                v8::Local<v8::Array> coordinates_array = Nan::New<v8::Array>(2);
-                coordinates_array->Set(0, Nan::New<v8::Number>(feature.coordinates.x)); // latitude
-                coordinates_array->Set(1, Nan::New<v8::Number>(feature.coordinates.y)); // longitude
-                geometry_obj->Set(Nan::New("coordinates").ToLocalChecked(), coordinates_array);
-                feature_obj->Set(Nan::New("geometry").ToLocalChecked(), geometry_obj);
-
-                // create properties object
-                v8::Local<v8::Object> properties_obj = Nan::New<v8::Object>();
-                for (auto const& prop : feature.properties_vector) {
-                    set_property(prop, properties_obj);
-                }
+        try {
+            Nan::HandleScope scope;
```

Comment:
@mapsam - the HandleScope needs to be outside of the try/catch. Otherwise it will not be able to function to clean up V8 objects in the `catch` case.

Reply:
good catch @springmeyer - https://github.com/mapbox/vtquery/pull/84